### PR TITLE
Added routing and better data fetching

### PR DESCRIPTION
**Pull request by JulianTapales**
- Content routing now works via aliases
- Basic list of working links rendered in reference page 
- Apollo now used instead of Lokka
- Refactor of directory structure
- Small style tweaks

### CODE REVIEW

File: src/components/DocsView/DocsView.tsx [hunk 1]
```diff
@@ -71,6 +28,7 @@ export default class DocsView extends React.Component<Props, State> {
       top: 0;
       left: 0;
       height: 100vh;
+
```
Comment (schickling) [inline src/components/DocsView/DocsView.tsx]: what's that newline for? :)

File: src/components/DocsView/components/ContentHandler.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+interface Props {
+  location: any
+  history: any
+  data: any
+  router: any
+}
+
+const getItem = gql`query getItem($alias: String) {
+    Item(alias: $alias) {
```
Comment (schickling) [inline src/components/DocsView/components/ContentHandler.tsx]: Indentation

File: src/components/DocsView/components/Header.tsx [hunk 3]
```diff
@@ -112,27 +109,33 @@ export default class Header extends React.Component<{}, {}> {
       }
     `
     let stateStyles = this.state.menuOpened ? styles.active : styles.inactive
```
Comment (schickling) [inline src/components/DocsView/components/Header.tsx]: always use `const` if possible

File: src/components/DocsView/components/TemporaryNavigation.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+export default class TemporaryNavigation extends React.Component<Props, {}> {
+
+  render() {
+    return (
+      <div>
+        <ul>
+          {this.props.links.map((link) => (
+            <li key={link.id}>
+              <Link to={link.path + '/' + link.alias}>{link.title}
```
Comment (schickling) [inline src/components/DocsView/components/TemporaryNavigation.tsx]: either have everything in one line or spread across multiple lines

File: src/components/DocsView/pages/ReferencePage.tsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+      <div>
+        <h1>Landing page for the references</h1>
+        {this.props.data.loading ? 'loading...' : <TemporaryNavigation links={this.props.data.allItems} />}
+      </div>
+    )
+  }
+}
+
+const getItem = gql`query getItemsByLayout($layout: ITEM_LAYOUT) {
+    allItems(filter: {layout: $layout}) {
```
Comment (schickling) [inline src/components/DocsView/pages/ReferencePage.tsx]: indentation (btw, you can change this in webstorm)

File: src/main.tsx [hunk 6]
```diff
@@ -6,32 +6,40 @@ import HomeView from './components/HomeView/HomeView'
...
+          <Route path='/about' component={AboutView}/>
+          <Route path='/faq' component={FAQ}/>
+          <Route path='/docs' component={DocsView}>
+            <Route path='quickstart' component={QuickstartPage}/>
+            <Route path='resources' component={ResourcesPage}/>
+            <Route path='reference' component={ReferencePage}/>
+            <Route path='blog' component={BlogPage}/>
+            <Route path='community' component={CommunityPage}/>
+          </Route>
+          <Route path='**' component={ContentHandler}></Route>
```
Comment (schickling) [inline src/main.tsx]: What does double star mean?

File: src/components/DocsView/pages/ReferencePage.tsx (hunk 5 above)
Comment (JulianTapales) [inline src/components/DocsView/pages/ReferencePage.tsx]: Got it - just changed it now

File: src/main.tsx (hunk 6 above)
Comment (JulianTapales) [inline src/main.tsx]: I thought it was for matching until the end of the URL, but on review one * is the correct selector :-) https://github.com/ReactTraining/react-router/blob/master/docs/guides/RouteMatching.md

File: src/components/DocsView/components/TemporaryNavigation.tsx (hunk 4 above)
Comment (JulianTapales) [inline src/components/DocsView/components/TemporaryNavigation.tsx]: 👍

File: src/components/DocsView/components/ContentHandler.tsx (hunk 2 above)
Comment (JulianTapales) [inline src/components/DocsView/components/ContentHandler.tsx]: Got it

File: src/components/DocsView/DocsView.tsx (hunk 1 above)
Comment (JulianTapales) [inline src/components/DocsView/DocsView.tsx]: Oops :-)

**Comment by marktani** [comment]
The amount of commits isn't too damn high :+1:

**Pull request merged**